zerver/lib/events.py

```diff
@@ -246,6 +242,9 @@ def fetch_initial_state_data(user_profile: UserProfile,
         else:
             state['realm_signup_notifications_stream_id'] = -1
 
+        if settings.VIDEO_ZOOM_CLIENT_ID is None or settings.VIDEO_ZOOM_CLIENT_SECRET is None:
+            del state['realm_available_video_chat_providers']['zoom']
```

Comment:
This deepcopy/remove block should probably instead just be modifying `VIDEO_CHAT_PROVIDERS` in `models.py` based on these settings.